rust/macros/module.rs

```diff
@@ -512,17 +514,28 @@ pub(crate) fn module(ts: TokenStream) -> TokenStream {
             static THIS_MODULE: kernel::ThisModule = unsafe {{ kernel::ThisModule::from_ptr(core::ptr::null_mut()) }};
 
             // Loadable modules need to export the `{{init,cleanup}}_module` identifiers
+            /// Tries to initialize the module and then unsynchronously sets `__MOD` to the initialized module.
+            /// Internally calls `__init()`.
+            ///
+            /// # Safety
+            ///
+            /// Should not be called unsynchronously with `cleanup_module()`, which unsets `__MOD`.
```

Comment:
This method also needs to be called before any function of the kernel module can be used.